### PR TITLE
Feature/job module operations

### PR DESCRIPTION
**Pull request by KashyretsIvanna**
feature: added endpoints: 

- checkChatOppened: check available conversations between freelancer and job owner and returns freelancer and array of this conversations 
- getUserJobs is improved and now it only returns jobs without any conversations with chosen freelancer
check: check this endpoints please

### CODE REVIEW

File: src/modules/user/user.service.ts [hunk 1]
```diff
@@ -268,33 +269,72 @@ export class UserService {
...
+        });
+      }
+
+      if (search) {
+        query.andWhere(
           new Brackets((qb) => {
             qb.where('user.category LIKE :search')
               .orWhere('user.position LIKE :search')
               .orWhere('user.hourly_rate LIKE :search')
               .orWhere('user.available_time LIKE :search', {
```
Comment (lyudad) [inline src/modules/user/user.service.ts]: Like is a very very expensive operation, it should be used only for search, in your case its filter

Comment (KashyretsIvanna) [reply src/modules/user/user.service.ts]: fixed

**Review by lyudad — APPROVED** [review]

File: src/modules/interview-invite/invite.service.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+export class InviteService {
+  constructor(
+    @InjectRepository(Conversation)
+    private conversationRepository: Repository<Conversation>,
+
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+  ) {}
+
+  async getUserById(id: number): Promise<User> {
```
Comment (SergKukh) [inline src/modules/interview-invite/invite.service.ts]: this method should be in user service

Comment (KashyretsIvanna) [reply src/modules/interview-invite/invite.service.ts]: fixed

**Review by SergKukh — APPROVED** [review]

File: src/modules/user/user.service.ts [hunk 3]
```diff
@@ -291,33 +292,74 @@ export class UserService {
...
+        page,
+        take,
+        skills,
+        categories,
+        hourly_rate_start,
+        hourly_rate_end,
+        search,
+        ...userPayload
+      } = params;
+      const role = 'Freelancer';
```
Comment (lyudad) [inline src/modules/user/user.service.ts]: why this constant here. I think you need to have some general constant, I don't think you use `Freelancer` only here

Comment (KashyretsIvanna) [reply src/modules/user/user.service.ts]: fixed

**Pull request merged**